components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/wireguard_tunnel_service.cc

```diff
@@ -0,0 +1,227 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/wireguard_tunnel_service.h"
+
+#include <windows.h>
+#include <winsvc.h>
+#include <string>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/command_line.h"
+#include "base/files/file_util.h"
+#include "base/files/scoped_temp_dir.h"
+#include "base/logging.h"
+#include "base/path_service.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/components/brave_vpn/browser/connection/common/win/scoped_sc_handle.h"
+#include "brave/components/brave_vpn/browser/connection/common/win/utils.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h"
+
+namespace brave_vpn {
+
+namespace {
+
+constexpr wchar_t kBraveWireguardConfig[] = L"wireguard.brave.conf";
+
+bool IsServiceRunning(SC_HANDLE service) {
+  SERVICE_STATUS service_status = {0};
+  if (!::QueryServiceStatus(service, &service_status)) {
+    return false;
+  }
+  return service_status.dwCurrentState == SERVICE_RUNNING;
+}
+
+}  // namespace
+
+namespace wireguard {
+
+// Created and launches a new Wireguard Windows service using passed config.
+// Before to start a new service it checks and removes existing if exists.
+bool LaunchWireguardService(const std::wstring& config) {
+  if (!RemoveExistingWireguradService()) {
+    VLOG(1) << "Failed to remove existing brave wiregurad service";
+    return false;
+  }
+  return CreateAndRunBraveWireguardService(config);
+}
+
+bool RemoveExistingWireguradService() {
```

Comment:
```suggestion
bool RemoveExistingWireguardService() {
```

Reply:
might want to search/replace the whole thing 😄

Reply:
done, thanks